include/bluetooth/services/dfu_smp_c.rst

```diff
@@ -0,0 +1,77 @@
+.. _dfu_smp_c_readme:
+
+GATT DFU SMP Service Client
+###########################
+
+This module implements a Simple Management Protocol (SMP) Service Client that can be used in the context of Device Firmware Updates (DFU).
+SMP is a basic transfer encoding for use with the `MCUmgr`_ management protocol.
+See `SMP over Bluetooth`_ for the service specification.
+
+The SMP Client module can be used to interact with Zephyr's :ref:`zephyr:smp_svr_sample`.
+
+The SMP Client implements only the service.
+It does not provide any functionality to process or interpret SMP commands and responses.
+
+Service UUID
+************
+
+The 128-bit service UUID is 8D53DC1D-1DB7-4CD3-868B-8A527460AA84 (16-bit offset: 0x0001).
+
+Characteristics
+***************
+
+This service has one characteristic that is used for both requests and responses.
+
+SMP Characteristic (DA2E7828-FBCE-4E01-AE9E-261174997C48)
+=========================================================
+
+Write Without Response
+   * Write an SMP command with data.
+
+Notify
+   * Response to the SMP command is sent using notification.
+   * Multiple notifications may be generated if the response does not fit in a single MTU.
+
+Usage
+*****
+
+.. note::
+   Do not access any of the values in the :cpp:type:`bt_gatt_dfu_smp_c` object structure directly.
+   All values that should be accessed have accessor functions.
+   The reason that the structure is fully defined is to allow the application to allocate the memory for it.
+
+MTU negotiation
+===============
+
+The current DFU SMP Server implementation in the :ref:`zephyr:smp_svr_sample` requires the whole message to be sent in one transfer.
+For most operations, this requires a bigger MTU size than the default one.
+Therefore, you must increase the default MTU size for the connection.
+Writing long characteristic values is not supported.
+
+Note that this is a limitation of the :ref:`zephyr:smp_svr_sample`, not of the SMP Client.
+
+Sending a command
+=================
+
+To send a command, use :cpp:func:`bt_gatt_dfu_smp_c_command`.
+The command is provided as a buffer.
+You can use :cpp:type:`dfu_smp_header` to simplify command header formatting.
```

Comment:
How?

Reply:
By casting the pointer to the buffer to the pointer of the dfu_smp_header structure type. I am not sure if we should explain such a things.

Reply:
Changed to "The command is provided as a buffer consisting of a :cpp:type:`dfu_smp_header` and the payload." - correct?

Reply:
"The command is provided as a **raw binary** buffer consisting of... To make it clear that we are not providing dfu_smp_header directly." But it is just a proposal. I am not 100% convinced if it is good.